src/testapp/sessions/solar_system.cpp

```diff
@@ -0,0 +1,492 @@
+
+    rUniverse.m_coordCommon.resize(rUniverse.m_coordIds.capacity());
+
+    CoSpaceCommon &rMainSpaceCommon = rUniverse.m_coordCommon[mainSpace];
+    rMainSpaceCommon.m_satCount     = c_planetCount;
+    rMainSpaceCommon.m_satCapacity  = c_planetCount;
+
+    Session out;
+    OSP_DECLARE_CREATE_DATA_IDS(out, topData, TESTAPP_DATA_SOLAR_SYSTEM_PLANETS);
+    auto& rCoordNBody = top_emplace< osp::KeyedVec<CoSpaceId, CoSpaceNBody> >(topData, idCoordNBody);
+    rCoordNBody.resize(rUniverse.m_coordIds.capacity());
+
+    // Associate each planet satellite with their surface coordinate space
+    for (SatId satId = 0; satId < c_planetCount; ++satId)
+    {
+        CoSpaceId const surfaceSpaceId = satSurfaceSpaces[satId];
+        CoSpaceCommon &rCommon = rUniverse.m_coordCommon[surfaceSpaceId];
+        rCommon.m_parent    = mainSpace;
+        rCommon.m_parentSat = satId;
+    }
+
+    // Coordinate space data is a single allocation partitioned to hold positions, velocities, and
+    // rotations.
+    // TODO: Alignment is needed for SIMD (not yet implemented). see Corrade alignedAlloc
+
+    std::size_t bytesUsed = 0;
+
+    // Positions and velocities are arranged as XXXX... YYYY... ZZZZ...
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satPositions[0]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satPositions[1]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satPositions[2]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satVelocities[0]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satVelocities[1]);
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satVelocities[2]);
+
+    // Rotations use XYZWXYZWXYZWXYZW...
+    partition(bytesUsed, c_planetCount, rMainSpaceCommon.m_satRotations[0],
+                                      rMainSpaceCommon.m_satRotations[1],
+                                      rMainSpaceCommon.m_satRotations[2],
+                                      rMainSpaceCommon.m_satRotations[3]);
+
+    partition(bytesUsed, c_planetCount, rCoordNBody[mainSpace].satOrbit);
+
+
+    // Allocate data for all planets
+    rMainSpaceCommon.m_data = Array<unsigned char>{Corrade::NoInit, bytesUsed};
+
+    // Create easily accessible array views for each component
+    auto const [x, y, z]        = sat_views(rMainSpaceCommon.m_satPositions,  rMainSpaceCommon.m_data, c_planetCount);
+    auto const [vx, vy, vz]     = sat_views(rMainSpaceCommon.m_satVelocities, rMainSpaceCommon.m_data, c_planetCount);
+    auto const [qx, qy, qz, qw] = sat_views(rMainSpaceCommon.m_satRotations,  rMainSpaceCommon.m_data, c_planetCount);
+
+    auto const nBodyView = rCoordNBody[mainSpace].satOrbit.view(arrayView(rMainSpaceCommon.m_data), c_planetCount);
+
+    x[Planets::SUN] = 0;
+    y[Planets::SUN] = 0;
+    z[Planets::SUN] = 0;
+    vx[Planets::SUN] = 0.0;
+    vy[Planets::SUN] = 0;
+    vz[Planets::SUN] = 0.0;
+
```

Comment:
I like how this is a fairly dead-simple way to assign a lot of data, though takes a lot of space as a tradeoff. Along with removing `enum Planets`, it may help to make a helper lambda:
```cpp
std::size_t nextBody = 0;
auto const add_body = [&rMainSpaceCommon, &rMainSpaceCommon, &nextBody] (Vector3l pos, Quaterion rot, float mass, Color4 color)
{
    auto const [x, y, z]        = sat_views(rMainSpaceCommon.m_satPositions,  rMainSpaceCommon.m_data, c_planetCount);
    auto const [vx, vy, vz]     = sat_views(rMainSpaceCommon.m_satVelocities, rMainSpaceCommon.m_data, c_planetCount);
    auto const [qx, qy, qz, qw] = sat_views(rMainSpaceCommon.m_satRotations,  rMainSpaceCommon.m_data, c_planetCount);
    
    x[nextBody] = pos.x();
    y[nextBody] = pos.y();
    ...
    ++nextBody
};

add_body(Vector3l{...}, {}, ....); // Sun
add_body(Vector3l{...}, {}, ....); // Blue planet
add_body(Vector3l{...}, {}, ....); // Red planet
```
Calling sat_view multiple times here as there are inconveniences with structured bindings and lambda captures, like [x, y, z]. They're technically not considered variables.